### ISSUE TITLE
Virtualenv activate scripts cause infinite loop in Powershell when multiple are activated

### ISSUE THREAD

**Issue by notskm** [conan-io/conan#5535]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

## Environment
OS: Windows 10
Conan version: 1.17.0
Python: 3.7.4
Powershell: 5.1

## Problem
Running multiple virtualenv scripts in Powershell causes an infinite loop where it repeatedly prints the prompt.

## Reproducing the problem
```
> conan install cmake_installer/3.15.0@conan/stable -g virtualenv -g virtualbuildenv
> ./activate_build.ps1
> ./activate.ps1
```
Output:
```
(conanenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv)
(conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) 
(conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) 
(conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) 
(conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) 
(conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) 
(conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) 
(conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) 
(conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) 
(conanbuildenv) (conanbuildenv) (conanbuildenv) (conanbuildenv) ...
```

**Comment by danimtb** [comment]
Hi @notskm and thanks a lot for reporting.

It looks like a clear bug that could be related to the variables where old environment is saved when the deactivate script is later executed. I would take a look at this. Have you found a workaround or any other hint about the reason of this issue?

Thanks!

**Comment by notskm** [comment]
As far as workarounds, you could either not use Powershell or edit the scripts manually. That's all I can think of. Unfortunately, the .sh scripts handle paths incorrectly #5528, so the only simple option is to use the .bat scripts.

I don't have any other information, unfortunately.

**Comment by notskm** [comment]
I just took a closer look at the activate scripts. The problem is that running multiple causes `_old_conan_prompt` to become recursive.

Each activate script begins with code to set modify Powershell's prompt function.

```powershell
function global:_old_conan_prompt {""}
$function:_old_conan_prompt = $function:prompt
function global:prompt { write-host "(conanenv) " -nonewline; & $function:_old_conan_prompt }
```

If we insert a couple lines, we can examine the content of those functions.
```powershell
echo "_old_conan_prompt: $function:_old_conan_prompt"
echo "global:prompt: $function:global:prompt"
```

Calling this script once works as expected and produces the following output:

```powershell
_old_conan_prompt:
"PS $($executionContext.SessionState.Path.CurrentLocation)$('>' * ($nestedPromptLevel + 1)) ";
# .Link
# https://go.microsoft.com/fwlink/?LinkID=225750
# .ExternalHelp System.Management.Automation.dll-help.xml

global:prompt:  write-host "(conanenv) " -nonewline; & $function:_old_conan_prompt
```

Calling this script a second time or running a second activate script with the same `echo` commands inserted gets stuck in the infinite loop I described after producing the following output:

```powershell
_old_conan_prompt:  write-host "(conanenv) " -nonewline; & $function:_old_conan_prompt
global:prompt:  write-host "(conanenv) " -nonewline; & $function:_old_conan_prompt
```

As you can see, _old_conan_prompt is now recursive. When Powershell runs `global:prompt` to print the prompt, it gets stuck running `_old_conan_prompt`.

**Comment by notskm** [comment]
Any news on this?

A workaround for anybody that doesn't want to use the `.bat` scripts:

```powershell
Get-ChildItem ./*activate*.ps1 | ForEach-Object {
    $content = (get-content -Path $_ | Select-String -Pattern '_old_conan_prompt' -NotMatch)
    Set-Content -Path $_ -Value $content
}
```

This snippet removes the problematic lines in the `.ps1` scripts, allowing them to work.

**Comment by danimtb** [comment]
Hi @notskm and sorry for the late response.

I tried the steps and managed to reproduce the issue. I am not a powershell expert at all, so it is hard to evaluate the fix you proposed, but if you feel it is a good solution maybe you could contribute that as a pull request and we will include in the next releases. Thanks!

**Comment by mjvankampen** [comment]
I think the simple solution here is to make ```_old_conan_prompt``` unique for each type of virtualenv, so ```_old_conan_activate_prompt```, ```_old_conan_activate_build_prompt``` and ```_old_conan_activate_run_prompt```

**Comment by memsharded** [comment]
Fixed by https://github.com/conan-io/conan/pull/8228, will be released in 1.33
